packages/ra-core/src/controller/usePrevNextController.ts

```diff
@@ -124,37 +125,76 @@ export const usePrevNextController = <RecordType extends RaRecord = any>(
         );
     }
 
-    const [storedParams] = useStore<StoredParams>(
+    const [storedParams] = useStore<Partial<ListParams>>(
         storeKey || `${resource}.listParams`,
         {
-            filter: {},
-            order: sort.order,
-            sort: sort.field,
+            filter: filterDefaultValues,
+            order: initialSort.order,
+            sort: initialSort.field,
         }
     );
 
-    const { data, error, isLoading } = useGetList<RecordType>(
+    const dataProvider = useDataProvider();
+    const queryClient = useQueryClient();
+    const pagination = { page: 1, perPage: limit };
+    const sort = {
+        field: storedParams.sort,
+        order: storedParams.order,
+    };
+    const filter = { ...storedParams.filter, ...permanentFilter };
+    const { meta, ...otherQueryOptions } = queryOptions;
+    const params = { pagination, sort, filter, meta };
+
+    // try to use data from the cache first
+    const queryData = queryClient.getQueryData<{
+        data: RaRecord[];
+        total: number;
+    }>([
         resource,
+        'getList',
         {
-            sort: {
-                field: storedParams.sort,
-                order: storedParams.order,
+            ...params,
+            pagination: {
+                page: storedParams.page,
+                perPage: storedParams.perPage,
```

Comment:
I'm not sure we should use `storedParams.perPage` here.
This parameter does not serve the same purpose than `limit`: 
- `storedParams.perPage` is coming from `<List>`, it use to display `n` items per page in a `<List>`.
- `limit` is made to limit the maximum result to fetch.

For example: if `storedParams.perPage` is set to 25, the `<List>` will show 25 items per page. But `<PrevNextButton>` will fetch only 25 items and the user will not be able to navigate beyond the twenty-fifth item although the item total is greater.

The story `WithLimit` shows the behavior described above.

Reply:
I'm using the pagination params from the store to look for cached data from the list view, to avoid fecthing if possible. It may happen that the current record id isn't in this list. In this case, the component does fetch data using the `limit` (see the useQuery further down)